### ISSUE TITLE
Battery Attributes not working

### ISSUE THREAD

**Issue by iamhuman2414** [kevinvincent/ha-wyzesense#227]
How do I fix the battery attributes? They just stopped working I'm not sure when it started.

**Comment by iamhuman2414** [comment]
@sweichbr

**Comment by sweichbr** [comment]
I found that you had to trigger each of the sensors. Probably because they don't report in unless there is an event (likely to save battery).

**Comment by onebyside** [comment]
> I found that you had to trigger each of the sensors. Probably because they don't report in unless there is an event (likely to save battery).

Can you paste your fix for this problem, been awhile since the batt reported in.

**Comment by jdubz666** [comment]
Hey,

I'm having the same issue with motion and contact sensors.

My sensors suddenly stopped responded so I removed and then added them back.  It worked however they came back without battery and RSI attributes.

I made sure the sensor were being triggered to initiate the attributes. And I checked back a day after to see if they are added after a delay.  No luck.

I tried changing /config/custom_components/wyzesense/ changed device_state_attributes to extra_state_attributes

Strangely this fixed for the motion sensor but not the contact sensors...

Any idea for how to fix battery for contact sensors?

**Comment by onebyside** [comment]
same issue here

Bell
SM-G996W
________________________________

**Comment by stewjoel** [comment]
For anyone still using this integration. I've updated the code to resolve all of the errors I found. You can try my new code if you'd like: https://github.com/stewjoel/ha-wyzesense/

**Comment by sweichbr** [comment]
Thank you for this…my integration has been broken for a while.

However, I am getting this when I download the integration:

“The version 8762ee3 for this integration can not be used with HACS.”

Any suggestions?



From: stewjoel ***@***.***>
Date: Sunday, November 3, 2024 at 6:11 PM
To: kevinvincent/ha-wyzesense ***@***.***>
Cc: sweichbr ***@***.***>, Mention ***@***.***>
Subject: Re: [kevinvincent/ha-wyzesense] Battery Attributes not working (Issue #227)

For anyone still using this integration. I've updated the code to resolve all of the errors I found. You can try my new code if you'd like: https://github.com/stewjoel/ha-wyzesense/

**Comment by onebyside** [comment]
Worked fine here, pasted it in and restarted.

**Comment by stanwebber** [comment]
> Thank you for this…my integration has been broken for a while. However, I am getting this when I download the integration: “The version [8762ee3](https://github.com/kevinvincent/ha-wyzesense/commit/8762ee33aea90f1609bd38c10c8d441ee8a884d2) for this integration can not be used with HACS.” Any suggestions?

yeah, i also get a similar repository download/update error after applying stewjoel's modifications to my fork's binary_sensor.py file. manually copying the new binary_sensor.py to the custom_components folder works just fine, but installing thru hacs seems to be broken now. i don't know enough about hacs to identify what the issue is.
